### PR TITLE
feat: add createSignalFromObservable utils

### PR DESCRIPTION
**Pull request by akumatus**

### CODE REVIEW

**Comment by akumatus** [comment]
* **#8088** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/8088?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @akumatus and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/affine/shared/src/utils/signal.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  const subscription = observable$.subscribe(value => {
+    newSignal.value = value;
+  });
+  return {
+    signal: newSignal,
+    cleanup: () => subscription.unsubscribe(),
+  };
+}
+
+export { type Signal } from '@lit-labs/preact-signals';
```
Comment (Saul-Mirone) [inline packages/affine/shared/src/utils/signal.ts:25]: Why do you need to export this type here?

Comment (akumatus) [reply packages/affine/shared/src/utils/signal.ts:25]: Sometimes it is necessary to explicitly declare the TypeScript type:

**Review by Saul-Mirone — APPROVED** [review]
LGTM

**Pull request merged**